aptos-move/aptos-vm-types/src/module_and_script_storage/module_storage.rs

```diff
@@ -1,9 +1,29 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
+use aptos_types::state_store::state_value::StateValueMetadata;
+use move_binary_format::errors::PartialVMResult;
+use move_core_types::{account_address::AccountAddress, identifier::IdentStr};
 use move_vm_runtime::{DummyCodeStorage, ModuleStorage};
 
 /// Represents module storage used by the Aptos blockchain.
-pub trait AptosModuleStorage: ModuleStorage {}
+pub trait AptosModuleStorage: ModuleStorage {
+    /// Returns the state value metadata of an associated with this module. The
+    /// error is returned if there is a storage error. If the module does not exist,
+    /// `None` is returned.
+    fn fetch_state_value_metadata(
```

Comment:
how do we make sure to cleanly separate the functionality and apis vs the executor view traits? do the module interfaces just move here?

Reply:
Yes! `ModuleView` is no longer needed, we have all the traits here, so this essentially merges executor traits into here. The reason is that we do not really need an adapter for modules, and we want to have a separate flow from data ideally. Block-STM will implement `AptosModuleStorage` directly.

Block-STM is generic, but I also previously added a trait method to convert addresses, module names into generic `K: ModulePath` (state keys) so we are good there as well.